File: 220/220-1.rb
N=gets.to_i
A=gets.split.map(&:to_i)

cnt=0
(N-2).times do |t|
  x,y,z=s=A.slice(t,3)
  next if s.uniq.size != 3 
  cnt+=1 if (x!=s.max && x!=s.min) || (z!=s.max && z!=s.min)
  #puts "t:#{t} x:#{x} y:#{y} z:#{z} max:#{s.max} min:#{s.min} cnt:#{cnt}"
end
p cnt

File: 220/220-1.php
<?php
$n=fgets(STDIN);
$a=split(' ', rtrim(fgets(STDIN)));

$cnt=0;
for($i=0; $i<sizeof($a)-2; $i++){
  list($x,$y,$z)=$s=array_slice($a,$i,3);
  if(sizeof(array_unique($s))!=3) continue;
  if(($x!=max($s) && $x!=min($s)) || ($z!=max($s) && $z!=min($s))) $cnt+=1;
  //echo "\$i:$i \$x:$x \$y:$y \$z:$z max:".max($a)." min:".min($a)." \$cnt:$cnt \n";
}
echo $cnt;

